cpp/dolfinx/fem/FiniteElement.h

```diff
@@ -115,37 +113,80 @@ class FiniteElement
                                           Eigen::Dynamic, Eigen::RowMajor>>& X,
       const Eigen::Tensor<double, 3, Eigen::RowMajor>& J,
       const Eigen::Ref<const Eigen::Array<double, Eigen::Dynamic, 1>>& detJ,
-      const Eigen::Tensor<double, 3, Eigen::RowMajor>& K,
-      const std::uint32_t permutation_info) const;
+      const Eigen::Tensor<double, 3, Eigen::RowMajor>& K) const;
 
-  /// Tabulate the reference coordinates of all dofs on an element
-  /// @return The coordinates of all dofs on the reference cell
-  ///
-  /// @note Throws an exception if dofs cannot be associated with points
-  const Eigen::Array<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>&
-  dof_reference_coordinates() const;
-
-  /// Map values of field from physical to reference space which has
-  /// been evaluated at points given by dof_reference_coordinates()
-  void transform_values(
-      ufc_scalar_t* reference_values,
-      const Eigen::Ref<const Eigen::Array<ufc_scalar_t, Eigen::Dynamic,
-                                          Eigen::Dynamic, Eigen::RowMajor>>&
-          physical_values,
-      const Eigen::Ref<const Eigen::Array<double, Eigen::Dynamic,
-                                          Eigen::Dynamic, Eigen::RowMajor>>&
-          coordinate_dofs) const;
-
-  /// Return the number of sub elements (for a mixed element)
-  int num_sub_elements() const;
+  /// Get the number of sub elements (for a mixed element)
+  /// @return the Number of sub elements
+  int num_sub_elements() const noexcept;
 
   /// Return simple hash of the signature string
-  std::size_t hash() const;
+  std::size_t hash() const noexcept;
 
   /// Extract sub finite element for component
   std::shared_ptr<const FiniteElement>
   extract_sub_element(const std::vector<int>& component) const;
 
+  /// Check if interpolation into the finite element space is an
+  /// identity operation given the evaluation on an expression at
+  /// specific points, i.e. the degree-of-freedom are equal to point
+  /// evaluations. The function will return `true` for Lagrange
+  /// elements.
+  ///  @return True is interpolation is an identity operation
+  bool interpolation_ident() const noexcept;
```

Comment:
This is long referred to as the element mapping being "identity", in UFL. Or do you mean something else?

Reply:
But what is 'the element mapping'?

This function returns true for a point evaluation element. I_h(u) = A u(x_i), where  I_h(u) are the dofs, u(x) is the function u evaluated at points {x_i} and A is the interpolation matrix. Function returns true if A = Identity.

Reply:
It is a map which tells how degrees of freedom transform under a coordinate transformation (reference to physical, or physical to reference, I don't remember the direction). This was included in transform dofs or similar in the past.

Why is this not fetched from the types of dofs then, if it should be all about point-eval dofs?

I think one should be able to show - at least for affine elements - that if all dofs are point-eval then they transform via the "identity".

Reply:
I think this is nicer than a check on dofs, which we don't have anyway in this class. This function just says that the interpolation operator is the identify, so no need to call the `interpolate` function. I could do something whacky and use 2\delta_{ij} for defining the dofs, in which case one would need to call the interpolate function as it would apply the scaled identity I/2.